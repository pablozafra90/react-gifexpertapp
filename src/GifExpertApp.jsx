import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

  const [repeated, setRepeated] = useState(false)

  const onNewCategory = ( newCategory ) => {

    const exist = categories.find( e => {
        return e.toLowerCase() === newCategory.toLowerCase()
    });

    if (exist) {
      //console.log('category already exist: ' + exist + ' === ' + newCategory);
      setRepeated(true);
      return;
    }

    setCategories([newCategory, ...categories]);
    setRepeated(false);
    //console.log('category is new: ' + exist + ' === ' + newCategory);
  }

  const [categories, setCategories] = useState([]);
  
  const clearAll = () => { setCategories([]); setRepeated(false); }

  return (
    <>
        <h1>¡Gracias por estar aquí!</h1>

        <h4>Esto es una práctica de React sencilla en la que el usuario obtiene 10 resultados de Giphy por cada input.<br/>Como extra, están capadas las duplicidades de búsquedas, aun diferenciándolas por mayúsculas o minúsculas.</h4>

        <div className="tool-content">

          {/* Input */}
          <AddCategory onNewCategory={ onNewCategory }/>
          
          <button type="submit" form="form01" value="{SubmitEvent}" className="buttons">Buscar</button>

          <button onClick={ clearAll } className="buttons">Borrar</button>

        </div>

        { repeated && (<h5 className="error-msg">Esta búsqueda está repetida!</h5>) }

        {/*listado de gifs*/}
        <ol>
          { categories.map( cat => (
                <GifGrid key={ cat } category={ cat }/>
              )
          ) }
        </ol>
        {/*Gif item*/}
    </>
  )
};
