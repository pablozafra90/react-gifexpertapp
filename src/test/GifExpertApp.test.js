import { render, screen } from "@testing-library/react";
import { GifExpertApp } from "../GifExpertApp";


describe('Test en <GifExpertApp/>', () => {

    test('Debe hacer match con snapshot', () => {
        
        const { container } = render( <GifExpertApp /> );
        expect(container).toMatchSnapshot();
        
    });
    
    test('Debe mostrar un título y un subtítulo', () => {
        
        render( <GifExpertApp /> );
        expect( screen.getAllByRole('heading').length ).toBe(2);
        //screen.debug();
        
    });
    test('Debe mostrar los botones de busca y borrar', () => {
        
        render( <GifExpertApp /> );
        expect( screen.getAllByRole('button').length ).toBe(2);
        expect( screen.getByText('Buscar') );
        expect( screen.getByText('Borrar') );
        //screen.debug();
        
    });

    test('Debe mostrar una lista vacia de categorias', () => {

        // GifExpertApp.mockReturnValue({
        //     categories: [ 'hola', ],
        //     repeated: true
        // })

        render( <GifExpertApp /> );
        //screen.debug();
        
    });
    
    
});
