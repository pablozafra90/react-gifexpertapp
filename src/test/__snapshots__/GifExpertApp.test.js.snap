// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <GifExpertApp/> Debe hacer match con snapshot 1`] = `
<div>
  <h1>
    ¡Gracias por estar aquí!
  </h1>
  <h4>
    Esto es una práctica de React sencilla en la que el usuario obtiene 10 resultados de Giphy por cada búsqueda.
    <br />
    Como extra, están capadas las duplicidades de búsquedas, aun diferenciándolas por mayúsculas o minúsculas.
  </h4>
  <div
    class="tool-content"
  >
    <form
      aria-label="form"
      id="form01"
    >
      <input
        placeholder="Buscar gifs"
        type="text"
        value=""
      />
    </form>
    <button
      class="buttons"
      form="form01"
      type="submit"
      value="{SubmitEvent}"
    >
      Buscar
    </button>
    <button
      class="buttons"
    >
      Borrar
    </button>
  </div>
  <ol />
</div>
`;
